### PR TITLE
feat(shell): hasfolder to support case insensitivity

### PR DESCRIPTION
**Pull request by warrenbuckley**
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [ ] Tests for the changes have been added (for bug fixes / features).
- [ ] Docs have been added/updated (for bug fixes / features).

### <samp>🤖 Generated by Copilot at 63ca563</samp>

Fix case-sensitive folder check for git segment on Windows. Use `filepath.Glob` and `strings.EqualFold` to compare folder names in `HasFolder` function in `src/platform/shell.go`.

### <samp>🤖 Generated by Copilot at 63ca563</samp>

*  Fix case-insensitive folder detection on Windows for git segment ([link](https://github.com/JanDeDobbeleer/oh-my-posh/pull/4308/files?diff=unified&w=0#diff-7b082f4396f0c936b17e8deb6c4ea8ec54cb48509691b08354c28ef55aa18557L499-R527))

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

**Comment by warrenbuckley** [comment]
@JanDeDobbeleer I am not sure I will be doing PRs for the remaining as I don't think they need it as much IMO, but if you feel otherwise, I am happy to update this PR

* HasParentFilePath
* HasFilesInDir
* HasFileInParentDirs (This just calls `HasFilesInDir`)

File: src/platform/shell.go [hunk 1]
```diff
@@ -496,14 +496,35 @@ func (env *Shell) HasFileInParentDirs(pattern string, depth uint) bool {
...
 	if err != nil {
 		env.Debug("false")
 		return false
 	}
-	isDir := f.IsDir()
-	env.DebugF("%t", isDir)
-	return isDir
+
+	for _, match := range matches {
+		if strings.EqualFold(filepath.Base(match), filepath.Base(folder)) {
```
Comment (JanDeDobbeleer) [inline src/platform/shell.go]: invert this with a continue. The entire block below shouldn't be indented that way.

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]
I wonder about the performance impact here. We didn't have a loop, now we do and I'm not certain we need this case-insensitive for all use-cases.

File: src/platform/shell.go (hunk 1 above)
Comment (warrenbuckley) [reply src/platform/shell.go]: @JanDeDobbeleer inverted with a continue as suggested

**Comment by warrenbuckley** [comment]
> I wonder about the performance impact here. We didn't have a loop, now we do and I'm not certain we need this case-insensitive for all use-cases.

@JanDeDobbeleer would you prefer a new method instead or a new argument. Just let me know what you prefer me to do here please :)

**Comment by JanDeDobbeleer** [comment]
Closing due to comment in main PR, keeping this for reference.

**Pull request closed without merging**